repo/blob/s3/s3_storage.go

```diff
@@ -312,6 +312,42 @@ func (s *s3Storage) FlushCaches(ctx context.Context) error {
 	return nil
 }
 
+func isAccessDenied(err error) bool {
+	var e minio.ErrorResponse
+	return errors.As(err, &e) && e.Code == "AccessDenied"
+}
+
+func (s *s3Storage) bucketExists(ctx context.Context) (bool, error) {
```

Comment:
Overall, it seems reasonable to do. The initial reaction or concern is about how this behaves with different S3 providers.

Let's make sure we provide information in the error that is useful to the user, and for debugging issues, etc.

Main comments are nits around some of the functionality and code structure.

- Make this a package (helper) function instead of a function with a receiver. Explicitly pass the `cli` client and the bucket name.
- Move (from the caller to this function) some of the error handling and wrapping.
- Return only error, instead of `bool, error`, simply check for error on the caller and propagate without wrapping at the caller

Reply:
I have implemented the 2nd and 3rd suggestions above, but having difficulties for the 1st one -- making `bucketExists` as a helper function.
`bucketExists` needs to call `s3Storage.GetBlob` or `s3Storage.getBlobWithVersion`, which further depends on some functions from `s3Storage`, i.e., `getObjectNameString`.
I tried to modify the code, it would either cause code redundancies or a chain of changes to these functions.

Therefore, can we leave the function with `s3Storage`? If the name `bucketExists` is more like a helper function, how about change its name, e.g., reachBucket?

Reply:
For now, I've changed the function name to `bucketAccessible`. Let me know for any other suggestions.

Reply:
nit: what about `checkBucketAccess` ?

Reply:
Done, change it to `checkBucketAccess`